gplugins/lumerical/utils.py

```diff
@@ -0,0 +1,154 @@
+from gdsfactory.technology import LayerStack
+from gdsfactory.typings import PathType
+from gdsfactory.pdk import get_layer_stack
+from gdsfactory.config import logger
+from xml.etree.ElementTree import Element, SubElement
+from xml.dom import minidom
+import xml.etree.ElementTree as ET
+from pathlib import Path
+
+um = 1e-6
+
+
+def to_lbr(
+    material_map: dict[str, str],
+    layerstack: LayerStack | None = None,
+    dirpath: PathType | None = "",
+) -> None:
+    """
+    Generate an XML file representing a Lumerical Layer Builder process file based on provided material map.
+
+    Args:
+        material_map: A dictionary mapping materials used in the layer stack to Lumerical materials.
+        layerstack: Layer stack that has info on layer names, layer numbers, thicknesses, etc.
+        dirpath: Directory to save process file (process.lbr)
+
+    Returns:
+        None
+
+    Notes:
+        This function generates an XML file representing a Layer Builder file for Lumerical, based on the provided the active PDK
+        and material map. It creates 'process.lbr' in the current working directory, containing layer information like name,
+        material, thickness, sidewall angle, and other properties specified in the layer stack. It skips layers that are not
+        of type 'grow' or 'background' and logs a warning for each skipped layer.
+    """
+    layerstack = layerstack or get_layer_stack()
+
+    layer_builder = Element("layer_builder")
+
+    process_name = SubElement(layer_builder, "process_name")
+    process_name.text = "process"
+
+    layers = SubElement(layer_builder, "layers")
+    doping_layers = SubElement(layer_builder, "doping_layers")
+    for layer_name, layer_info in layerstack.to_dict().items():
+        ### Set optical and metal layers
+        if layer_info["layer_type"] == "grow":
+            process = "Grow"
+        elif layer_info["layer_type"] == "background":
+            process = "Background"
+        elif layer_info["layer_type"] == "doping":
+            process = "Implant"
+        else:
+            logger.warning(
+                f'"{layer_info["layer_type"]}" layer type not supported for "{layer_name}" in Lumerical. Skipping in LBR process file generation.'
+            )
+            process = "Grow"
+
+        if process == "Grow" or process == "Background":
+            layer = SubElement(layers, "layer")
+
+            # Default params
+            layer.set("enabled", "1")
```

Comment:
Maybe worth putting this in a dict somewhere else and just mapping layer.set over the items to improve readability, your call though